samples/JobsAgent/JobsAgent.cpp

```diff
@@ -0,0 +1,291 @@
+            return rc;
+        }
+
+        void JobsAgent::StartInstalledPackages() {
+            AWS_LOG_INFO(LOG_TAG_JOBS_AGENT, "StartInstalledPackages");
+
+            util::Map<util::String, util::String> statusDetailsMap;
+
+            for (util::JsonValue::ConstMemberIterator package_itr = installed_packages_json_.MemberBegin();
+                package_itr != installed_packages_json_.MemberEnd(); ++package_itr) {
+                if (package_itr->value.IsObject()) {
+                    util::String packageName = package_itr->name.GetString();
+                    if (PackageIsAutoStart(packageName.c_str())) {
+                        StartPackage(statusDetailsMap, packageName.c_str());
+                    }
+                }
+            }
+        }
+
+        ResponseCode JobsAgent::RunAgent(util::String processTitle) {
+            process_title_ = processTitle;
+
+            ResponseCode rc = util::JsonParser::InitializeFromJsonFile(installed_packages_json_, JobsAgent::INSTALLED_PACKAGES_FILENAME);
+            if (ResponseCode::FILE_OPEN_ERROR == rc) {
+                AWS_LOG_INFO(LOG_TAG_JOBS_AGENT, "Unable to open installed packages file %s, assuming no packages installed.", "test");//JobsAgent::INSTALLED_PACKAGES_FILENAME);
+                rc = util::JsonParser::InitializeFromJsonString(installed_packages_json_, "{}");
+                if (ResponseCode::SUCCESS != rc) {
+                    AWS_LOG_ERROR(LOG_TAG_JOBS_AGENT, "Unexpected initialization error: %s", ResponseHelper::ToString(rc).c_str());
+                }
+            } else if (ResponseCode::SUCCESS != rc) {
+                AWS_LOG_ERROR(LOG_TAG_JOBS_AGENT,
+                              "Error in Parsing. %s\n parse error code : %d, offset : %d",
+                              ResponseHelper::ToString(rc).c_str(),
+                              static_cast<int>(util::JsonParser::GetParseErrorCode(installed_packages_json_)),
+                              static_cast<unsigned int>(util::JsonParser::GetParseErrorOffset(installed_packages_json_)));
+                return rc;
+            }
+
+            rc = InitializeTLS();
+            if (ResponseCode::SUCCESS != rc) {
+                return rc;
+            }
+
+            ClientCoreState::ApplicationDisconnectCallbackPtr p_disconnect_handler =
+                std::bind(&JobsAgent::DisconnectCallback, this, std::placeholders::_1, std::placeholders::_2);
+
+            ClientCoreState::ApplicationReconnectCallbackPtr p_reconnect_handler =
+                std::bind(&JobsAgent::ReconnectCallback, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3);
+
+            ClientCoreState::ApplicationResubscribeCallbackPtr p_resubscribe_handler =
+                std::bind(&JobsAgent::ResubscribeCallback, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3);
+
+            p_iot_client_ = std::shared_ptr<MqttClient>(MqttClient::Create(p_network_connection_,
```

Comment:
I really really hope MqttClient::Create uses std new here instead of a different allocator, otherwise the deleters won't match.

We doesn't MqttClient::Create not used std::make_shared?

Reply:
This is consistent with the other sample applications. Making changes to MqttClient is probably outside the scope of this pull request.